### PR TITLE
Convert Ember.runInDebug and Ember.assert to modern syntax.

### LINKED ISSUE #241: runInDebug / asserts not stripped

### PR DESCRIPTION
**Pull request by lbdm44**
This PR converts the usage of `Ember.runInDebug` and _most_ Ember.* references to the modules syntax via `ember-modules-codemod`.

Fixes #241

### CODE REVIEW

File: MODULE_REPORT.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+## Module Report
```
Comment (rwjblue) [inline MODULE_REPORT.md]: I'd rather not commit this file, would you mind removing?

File: addon/resolvers/classic/index.js [hunk 2]
```diff
@@ -1,6 +1,14 @@
 /* globals requirejs, require */
 
+import { deprecate } from '@ember/application/deprecations';
```
Comment (rwjblue) [inline addon/resolvers/classic/index.js]: Can you swap this to `import { deprecate } from '@ember/debug'` instead?

File: addon/resolvers/classic/index.js [hunk 3]
```diff
@@ -23,29 +31,6 @@ export class ModuleRegistry {
   }
 }
 
-/*
```
Comment (rwjblue) [inline addon/resolvers/classic/index.js]: Can you bring this comment back (the codemod likely removed it)

File: index.js [hunk 4]
```diff
@@ -31,7 +31,7 @@ module.exports = {
       plugins: [
         [require('babel-plugin-debug-macros').default, {
           debugTools: {
-            source: '@ember/debug'
+            source: 'this-is-dumb-it-should-not-be-required-i-blame-rwjblue'
```
Comment (rwjblue) [inline index.js]: 👍 😝

File: tests/unit/resolvers/classic/basic-test.js [hunk 5]
```diff
@@ -19,31 +22,37 @@ function setupResolver(options) {
...
+
+    this.sandbox.restore();
   }
 });
 
 test("can access at deprecated 'resolver' module name", function(assert){
   assert.expect(2);
 
-  Ember.deprecate = function(message, test) {
+  deprecate = function(message, test) {
```
Comment (rwjblue) [inline tests/unit/resolvers/classic/basic-test.js]: Hmm, in general, I'm not sure this works (if it does its essentially a bug in babel-plugin-ember-modules-api-polyfill).  Instead we should use [ember-qunit-assert-helpers](https://github.com/workmanw/ember-qunit-assert-helpers) which provides an `assert.expectDeprecation` system that is pretty handy...

File: tests/unit/resolvers/classic/basic-test.js [hunk 6]
```diff
@@ -265,12 +274,12 @@ test('warns if looking up a camelCase helper that has a dasherized module presen
     return { default: { isHelperInstance: true } };
   });
 
-  Ember.warn = function(message, test, options) {
+  this.sandbox.stub(Ember, 'warn').callsFake((message, test, options) => {
```
Comment (rwjblue) [inline tests/unit/resolvers/classic/basic-test.js]: I think we can also replace this with `assert.expectWarning`

Comment (lbdm44) [reply tests/unit/resolvers/classic/basic-test.js]: Refactored this, it is much cleaner with those helpers. One thing I found with the helper is we are not able to reach in and assert against the warning's id.

**Comment by rwjblue** [comment]
Thank you!

**Pull request merged**